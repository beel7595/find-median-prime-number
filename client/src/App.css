.App {

  margin: 0px;
}

header{
  text-align: center;
  color: black;
  margin: 0px;

}

#form{
  width: 540px;
  margin: auto;
  border-radius: 4px;
  background-color: lightblue;
  min-height: 50vh;
  text-align: center;
  border-radius: 5px;
  box-shadow: 5px 5px 5px gray;
}

#numberInput{
  margin-top: 5rem;
  margin-bottom: 10px;
  height: 30px;
  border-radius: 5px;
  width: 300px;
  border-color: gray;
}



#submit{
  margin: auto;
  display: block;
  border-radius: 5px;
  background-color:lightsalmon;
  font-weight: bold;
  width: 100px;
  height: 25px;
}

#result{
  text-align: left;
  margin: 10px;
  padding-left: 100px;
  margin-top: 30px;
}

#result :nth-child(2){
  color: red;
  margin-left: 10px;
}

#hint{
  margin-bottom: 30px;
  font-size: 12px;
  color: red;
  margin-top: 0px;
}