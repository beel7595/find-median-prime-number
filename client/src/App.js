import axios from "axios"
import { useState } from 'react'
import "./App.css";

function App() {

  const [number, setNumber] = useState(0);
  const [result, setResult] = useState("");
  const [showHint, setShowHint] = useState(false);


  const API_URL = "/api/getResult";
  const IMPUT_LMIT = Math.pow(2, 32);

  const handleClick = (e) => {

    axios.post(API_URL, { number: number })
      .then((res) => {
        const { code, data, msg } = res.data;
        if (code === 0) {
          setResult('[' + data.toString() + ']');
        } else {
          setResult(msg);
        }
        console.log(res);
      })
      .catch((err) => {
        console.log(err);
      });
  }


  return (
    <div data-testid="app" className="App">
      <header>
        <h2>Input an upper limit number</h2>
        <h2>you will get a median prime number array of the set of prime numbers less than n</h2>
      </header>
      <div id="form">
        <input data-testid="input" id='numberInput' onChange={(event) => {
          if (!/[0-9]/.test(event.key)) {
            event.preventDefault();
          }
          if (event.target.value > IMPUT_LMIT) {
            setNumber(IMPUT_LMIT);
            setShowHint(true);
            return;
          }
          const targetValue = event.target.value;
          event.target.value = targetValue === '' ? 0 : parseInt(event.target.value);
          setNumber(event.target.value);
          setShowHint(false);
          setResult("");
        }} placeholder="please enter a numer..." type="number" value={number} min="1" max="5"></input>
        {
          showHint ? (<p id='hint'>Hint:The input number cannot exceed the 32nd power of 2</p>) : null
        }
        <input data-testid="submit" id='submit' type='submit' onClick={handleClick}></input>
        <div id='result'>
          <span>Result:</span>
          <span>{result}</span>
        </div>
      </div>

    </div >
  );
}

export default App;
