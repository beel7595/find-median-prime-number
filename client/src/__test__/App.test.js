import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer'

import App from '../App';


test('should render App', () => {
  render(<App />);
  const linkElement = screen.getByTestId("app");
  expect(linkElement).toBeInTheDocument();
})

test('matches snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
})

const stepUp = () => {
  const utils = render(<App />);
  const input = utils.getByTestId("input");
  const submit = utils.getByTestId("submit");
  return ({
    input,
    submit,
    utils
  })
}

test('input should be a number', () => {
  const { input } = stepUp();
  fireEvent.change(input, { target: { value: 'test' } });
  expect(input.value).toBe('0');
})

test('input should be a number without 0 ahead', () => {
  const { input } = stepUp();
  fireEvent.change(input, { target: { value: '0001' } });
  expect(input.value).toBe('1');
})

test('input should be a integer', () => {
  const { input } = stepUp();
  fireEvent.change(input, { target: { value: '100.54' } });
  expect(input.value).toBe('100');
})

test('input should be a number less than Less than 2 to the 32nd power', () => {
  const { input } = stepUp();
  fireEvent.change(input, { target: { value: '4294967297' } });
  expect(input.value).toBe('4294967296');
})

// test('input number to submit', () => {
//   const { utils, submit, input } = stepUp();
//   fireEvent.change(input, { target: { value: '100' } });
//   fireEvent.click(submit);
//   expect(utils.number).toBe('100');
// })
