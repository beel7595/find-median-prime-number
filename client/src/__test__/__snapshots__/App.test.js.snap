// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
  data-testid="app"
>
  <header>
    <h2>
      Input an upper limit number
    </h2>
    <h2>
      you will get a median prime number array of the set of prime numbers less than n
    </h2>
  </header>
  <div
    id="form"
  >
    <input
      data-testid="input"
      id="numberInput"
      max="5"
      min="1"
      onChange={[Function]}
      placeholder="please enter a numer..."
      type="number"
      value={0}
    />
    <input
      data-testid="submit"
      id="submit"
      onClick={[Function]}
      type="submit"
    />
    <div
      id="result"
    >
      <span>
        Result:
      </span>
      <span>
        
      </span>
    </div>
  </div>
</div>
`;
